product_harmonized_system_country/models/product_product.py

```diff
@@ -0,0 +1,20 @@
+# Copyright 2024 Manuel Regidor <manuel.regidor@sygel.es>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    def get_hs_local_code_by_country(self, country=False):
+        self.ensure_one()
+        hs_code = self.get_hs_code_recursively()
+        code = ""
+        if hs_code and country:
+            country_hs_code = hs_code.hs_code_country_ids.filtered(
+                lambda a: country.id in a.country_ids.ids
+            )
+            if len(country_hs_code) == 1:
```

Comment:
If there are multiple, the system does not return any error or inform the user?

Reply:
There should not be multiple instances as there is a restrictions. I thought of adding a warning, but I found it pointless.

Reply:
The system does not allow multiple assignment?
The warning is to differentiate if the system does not find anything or if it is a multiple assignment error. If it has been contracted elsewhere, no problem. Let me know.

Reply:
No, the system does not allow to have multiple country lines with the same country for the same HS code. There is a restriction for that, so the if statement here is just an extra check for something that should not happen unless the database is written straightaway. The restriction _check_country_ids in hs.code.contry, which we discussed last week, does that.

Reply:
perfect, thanks.